VTUL/iawa
# Fix the bug when the basepath is read from environment variable in th…

…e config file

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-1691) (:star:)

* Other Relevant Links:
This also fix the problem in https://github.com/VTUL/iawa/pull/119 with respect to the failing rspec test on reading the batch import basepath from configuration file.

# What does this Pull Request do? (:star:)
This PR fixes a bug when the basepath specified in the configuration file is read from either a string or an environment variable, ERB template engine is used to interpret the "<%= %>" ruby code block.

# What's the changes? (:star:)

* Added ERB to interpret the ruby code;
* Added deep_symbolize_keys to convert string keys to symbol keys;
* Turned on the rspec example for reading basepath from config file.

# How should this be tested?

* Test this in master branch first: try to change the basepath in either installscripts or config/batch_import.yml (this is easier). You can see that the basepath is always set to the default;
* Check out LIBTD-1691 branch: try to change the basepath as above. You can see that basepath can be changed accordingly;
* Test if batch import function works as expected;
* Run bin/rspec to see all the tests are passed.

# Interested parties
Tag (@shabububu ) interested parties

(:star:) Required fields
